### PR TITLE
Clean up rechargers and add wall recharger panel sprite

### LINKED ISSUE #18470: Rechargers open panel state switches back to its blinking even if the panel is still open
**Issue Description**:

The rechargers open panel state switches back to its default blinking state within seconds, even if the panel is still screwdrivered open

**What did you expect to happen**: 

For the recharger not to switch back to its blinking state whilst the panel is still screwdrivered open

**What happened instead**:

The rechargers open panel state was replaced with its default blinking animation state while the panel was still open

**Why is this bad/What are the consequences**:

Animations borked

**Steps to reproduce the problem**:

Screwdriver a recharger open and its open panel icon will quickly be replaced by the default blinking one even if the panel is still open

**When did the problem start happening**:

Noticed this on an up to date local branch

**Extra information**:

### PR DESCRIPTION
**Pull request by Vi3trice**
## What Does This PR Do
Cleans up the code for rechargers and make it more consistent in the process. They'd also constantly update the icon even when the panel was open when processing.

This also adds a check for the panel being open when trying to wrench it, to prevent a case where you can end up with an unanchored machine frame from just crowbaring the thing.

## Why It's Good For The Game
Fixes #18470.

## Images of changes
Charger_Panel
Recharger_Loaded

## Changelog
:cl:
tweak: Added feedback for trying to insert in an already loaded recharger
tweak: Unanchored rechargers can be screwed open or closed
fix: Rechargers properly show an open panel when using a screwdriver
imageadd: Wall recharger open panel sprite
/:cl:

### CODE REVIEW

File: code/game/machinery/recharger.dm [hunk 1]
```diff
@@ -34,44 +31,60 @@
...
+		return
+	if(panel_open)
+		to_chat(user, "<span class='warning'>Close the maintenance panel first!</span>")
+		return
+	if(charging)
+		to_chat(user, "<span class='warning'>There's \a [charging] inserted in [src] already!</span>")
+		return
+
+	//Checks to make sure he's not in space doing it, and that the area got proper power.
+	var/area/a = get_area(src)
```
Comment (AffectedArc07) [inline code/game/machinery/recharger.dm:51]:
```suggestion
	var/area/A = get_area(src)
```

File: code/game/machinery/recharger.dm [hunk 2]
```diff
@@ -34,44 +31,60 @@
...
+	if(!isarea(a) || !a.power_equip)
```
Comment (AffectedArc07) [inline code/game/machinery/recharger.dm:52]:
```suggestion
	if(!istype(A) || !A.power_equip)
```
You already casted it to an area in the above

File: code/game/machinery/recharger.dm [hunk 3]
```diff
@@ -34,44 +31,60 @@
...
+
+	if(istype(G, /obj/item/gun/energy))
+		var/obj/item/gun/energy/E = G
+		if(!E.can_charge)
+			to_chat(user, "<span class='notice'>Your gun has no external power connector.</span>")
+			return
+
+	if(!user.drop_item())
+		return
+	G.forceMove(src)
```
Comment (AffectedArc07) [inline code/game/machinery/recharger.dm:64]:
```suggestion
		return

	G.forceMove(src)
```
Readability

File: code/game/machinery/recharger.dm [hunk 4]
```diff
@@ -34,44 +31,60 @@
...
+		to_chat(user, "<span class='warning'>Remove the charging item first!</span>")
+		return
+	if(!I.use_tool(src, user, 0, volume = I.tool_volume))
+		return
+	panel_open = !panel_open
+	if(panel_open)
+		SCREWDRIVER_OPEN_PANEL_MESSAGE
+	else
+		SCREWDRIVER_CLOSE_PANEL_MESSAGE
+	update_icon()
```
Comment (AffectedArc07) [inline code/game/machinery/recharger.dm:87]:
```suggestion
	if(charging)
		to_chat(user, "<span class='warning'>Remove the charging item first!</span>")
		return

	if(!I.use_tool(src, user, 0, volume = I.tool_volume))
		return

	panel_open = !panel_open
	if(panel_open)
		SCREWDRIVER_OPEN_PANEL_MESSAGE
	else
		SCREWDRIVER_CLOSE_PANEL_MESSAGE
	
	update_icon()
```
More readability

**Review by AffectedArc07 — DISMISSED** [review]
Very minor nitpicks

**Comment by Bm0n** [comment]
I think you removed the anchored check on the screwdriver act meaning you can now get unanchored machine frames and half built things. not normally something you can do on paradise

File: code/game/machinery/recharger.dm [hunk 5]
```diff
@@ -34,47 +31,64 @@
...
+		to_chat(user, "<span class='warning'>Close the maintenance panel first!</span>")
+		return
+	if(charging)
+		to_chat(user, "<span class='warning'>There's \a [charging] inserted in [src] already!</span>")
+		return
+
+	//Checks to make sure he's not in space doing it, and that the area got proper power.
+	var/area/A = get_area(src)
+	if(!istype(A) || !A.power_equip)
+		to_chat(user, "<span class='notice'>[src] blinks red as you try to insert [G].</span>")
```
Comment (farie82) [inline code/game/machinery/recharger.dm:53]:
```suggestion
		to_chat(user, "<span class='warning'>[src] blinks red as you try to insert [G].</span>")
```

File: code/game/machinery/recharger.dm [hunk 6]
```diff
@@ -34,47 +31,64 @@
 		recharge_coeff = C.rating
 
 /obj/machinery/recharger/attackby(obj/item/G, mob/user, params)
+	. = TRUE
```
Comment (farie82) [inline code/game/machinery/recharger.dm:34]: Prob best to set this after the `allowed` check

File: code/game/machinery/recharger.dm [hunk 7]
```diff
@@ -34,47 +31,64 @@
...
-			var/area/a = get_area(src)
-			if(!isarea(a) || !a.power_equip)
-				to_chat(user, "<span class='notice'>[src] blinks red as you try to insert [G].</span>")
-				return TRUE
-
-			if(istype(G, /obj/item/gun/energy))
-				var/obj/item/gun/energy/E = G
-				if(!E.can_charge)
-					to_chat(user, "<span class='notice'>Your gun has no external power connector.</span>")
-					return TRUE
```
Comment (farie82) [inline code/game/machinery/recharger.dm:54]: You removed this check

**Review by farie82 — CHANGES_REQUESTED** [review]

**Review by farie82 — APPROVED** [review]

**Pull request merged**